adjacent_material.md

```diff
@@ -0,0 +1,233 @@
+highly recommend their presentations, the quality is just incredible!
+
+-   3Blue1Brown
+
+[But what is a neural network? \| Chapter 1, Deep
+learning](https://www.youtube.com/watch?v=aircAruvnKk)
+
+[Gradient descent, how neural networks learn \| Chapter 2, Deep
+learning](https://www.youtube.com/watch?v=IHZwWFHWa-w)
+
+[What is backpropagation really doing? \| Chapter 3, Deep
+learning](https://www.youtube.com/watch?v=Ilg3gGewQ5U)
+
+[Backpropagation calculus \| Chapter 4, Deep
+learning](https://www.youtube.com/watch?v=tIeHLnjs5U8)
+
+[But what is a
+convolution?](https://www.youtube.com/watch?v=KuXjwB4LzSA)
+
+[Convolutions \| Why X+Y in probability is a beautiful
+mess](https://www.youtube.com/watch?v=IaSGqQa5O-M)
+
+-   How to Create a Neural Network (and Train it to Identify Doodles)
+
+This course explores how neural networks learn by programming one from
+scratch in C\#, and then attempting to teach it to recognize various
+doodles and images. From a developer standpoint I believe that the
+approach used here has lots of merits.
+
+[How to Create a Neural Network (and Train it to Identify
+Doodles)](https://www.youtube.com/watch?v=hfMk-kjRv4c)
+
+## Stable Diffusion
+
+In addition to Jeremy Howard's explanation we found that it was worth
+giving the below course a try. It expresses in different words -- and
+examples - the process of image generation with the same core concepts
+introduced by Jeremy Howard (convolutional layer, U-Nets, autoencoders,
+etc).
+
+[How Stable Diffusion Works (AI Image
+Generation)](https://www.youtube.com/watch?v=sFztPP9qPRc)
+
+Some of us worked on the Stable Diffusion model via the DiffEdit
+implementation. The following are some of the papers we found very
+valuable.
+
+[DiffEdit Paper](https://arxiv.org/pdf/2210.11427.pdf)
+
+[Stable Diffusion with
+Diffusers](https://huggingface.co/blog/stable_diffusion#how-does-stable-diffusion-work)
+
+[How does Stable Diffusion
+work?](https://stable-diffusion-art.com/how-stable-diffusion-work/)
+
+[How does negative prompt
+work?](https://stable-diffusion-art.com/how-negative-prompt-work/)
+
+As mentioned earlier, we also created our own implementation of the
+DiffEdit technique and explained the process in a blog, which you can
+find here:
+
+[DiffEdit: Editing Images using Generative
+AI](https://blog.scottlogic.com/2023/12/13/diffedit.html)
+
+## Coding challenges
+
+Lastly, I think you can try the following coding challenges once you
+have a full knowledge of the Python libraries and deep learning models.
+The contents are presented in a very entertaining and educational way.
+
+[Building a Neural Network with PyTorch in 15 Minutes \| Coding
+Challenge](https://www.youtube.com/watch?v=mozBidd58VQ)
+
+[Building the Gradient Descent Algorithm in 15 Minutes \| Coding
+Challenge](https://www.youtube.com/watch?v=Souzjv6WfrY)
+
+[Building a ML Text to Image App with Stable Diffusion in 15 Minutes \|
+Coding Challenge](https://www.youtube.com/watch?v=7xc0Fs3fpCg)
```

Comment:
It's always worth adding a conclusion to a blog post to help the reader understand that it's the end of the post. A one sentence summary of the post, followed by some personal take on the content and a call to action is normally a decent formula.